docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,328 @@
+
+**Examples**
+
+```sh
+testcafe chrome my-tests --cs assets/jquery.js
+```
+
+```js
+runner.clientScripts('assets/jquery.js');
+// or
+runner.clientScripts({ path: 'assets/jquery.js' });
+```
+
+```js
+{
+    "clientScripts": "assets/jquery.js",
+    // or
+    "clientScripts": { "path": "assets/jquery.js" }
+}
+```
+
+> You cannot combine the `path`, [module](#inject-a-module) and [content](#inject-script-code) properties.
+
+### Inject a Module
+
+Specify the module name to inject the module's content into the tested pages. Use a string or an object with the `module` property.
+
+*CLI*
+
+```text
+testcafe <browser> <tests> --cs <module>
+```
+
+*Programming interface*
+
+```text
+runner.clientScripts( module | { module } )
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": "module" | { module }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts( module | { module } )
+test.clientScripts( module | { module } )
+```
+
+Argument  | Type   | Description
+--------- | ------ | ----------------
+`module`  | String | The module name.
+
+TestCafe searches for the module's entry point with Node.js mechanisms and injects its content into the tested page.
+
+**Examples**
+
+```sh
+testcafe chrome my-tests --cs lodash
+```
+
+```js
+fixture `My fixture`
+    .page `https://example.com`
+    .clientScripts('lodash');
+
+// or
+
+fixture `My fixture`
+    .page `https://example.com`
+    .clientScripts({ module: 'lodash' });
+```
+
+```js
+{
+    "clientScripts": "lodash",
+    // or
+    "clientScripts": { "module": "lodash" }
+}
+```
+
+> You cannot combine the `module`, [path](#inject-a-javascript-file) and [content](#inject-script-code) properties.
+
+### Inject Script Code
+
+You can provide the injected script as a string with JavaScript code. Pass an object with the `content` property to do this.
+
+*Programming interface*
+
+```text
+runner.clientScripts({ content })
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": { content }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts({ content })
+test.clientScripts({ content })
+```
+
+Argument  | Type   | Description
+--------- | ------ | ----------------------------------------
+`content` | String | JavaScript code that should be injected.
+
+**Examples**
+
+```js
+const mockDate = `
+    Date.prototype.getTime = function () {
+        return 42;
+    };
+`;
+
+test
+    ('My test', async t => { /* ... */ })
+    .clientScripts({ content: mockDate });
+```
+
+```json
+{
+    "clientScripts": {
+        "content": "Date.prototype.getTime = () => 42;"
+    }
+}
+```
+
+> You cannot combine the `content`, [path](#inject-a-javascript-file) and [module](#inject-a-module) properties.
+
+### Provide Scripts for Specific Pages
+
+You can also specify pages into which a script should be injected. This is helpful when you need to use custom mocks on particular pages and preserve the default behavior everywhere else.
```

Comment:
By default the script inject for pages visited during test execution. You can specify the particular pages into which...

Reply:
We can't use "*by default*" anywhere.